TDataFrame.hxx

```diff
@@ -22,828 +22,1042 @@
+   TActionResultPtr<COLL> Get(const std::string &branchName = "")
+   {
+      auto df = GetDataFrameChecked();
+      unsigned int nSlots = df.lock()->GetNSlots();
+      auto theBranchName(branchName);
       GetDefaultBranchName(theBranchName, "get the values of the branch");
-      TActionResultPtr<COLL> values (std::make_shared<COLL>(), fDerivedPtr->GetDataFrame());
-      auto valuesPtr = values.getUnchecked();
-      auto getAction = [valuesPtr](const T& v) { valuesPtr->emplace_back(v); };
+      auto valuesPtr = std::make_shared<COLL>();
+      TActionResultPtr<COLL> values(valuesPtr, df);
+      auto getOp = std::make_shared<Internal::Operations::GetOperation<T,COLL>>(valuesPtr, nSlots);
+      auto getAction = [getOp] (unsigned int slot , const T &v) mutable { getOp->Exec(v, slot); };
       BranchList bl = {theBranchName};
-      using DFA = TDataFrameAction<decltype(getAction), Derived>;
-      Book(std::unique_ptr<DFA>(new DFA(getAction, bl, *fDerivedPtr)));
+      using DFA_t = Internal::TDataFrameAction<decltype(getAction), Proxied>;
+      Book(std::shared_ptr<DFA_t>(new DFA_t(getAction, bl, fProxiedPtr)));
       return values;
    }
 
-   template<typename T = double>
-   TActionResultPtr<TH1F> Histo(const std::string& branchName, const TH1F& model) {
-      auto theBranchName (branchName);
+   template <typename T = Double_t>
+   TActionResultPtr<TH1F> Histo(const std::string &branchName, const TH1F &model)
+   {
+      auto theBranchName(branchName);
       GetDefaultBranchName(theBranchName, "fill the histogram");
       auto h = std::make_shared<TH1F>(model);
-      return CreateAction<T, EActionType::kHisto1D>(theBranchName,h);
+      return CreateAction<T, Internal::EActionType::kHisto1D>(theBranchName, h);
    }
 
-   template<typename T = double>
-   TActionResultPtr<TH1F> Histo(const std::string& branchName = "", int nBins = 128, double minVal = 0., double maxVal = 0.) {
-      auto theBranchName (branchName);
+   template <typename T = Double_t>
+   TActionResultPtr<TH1F> Histo(const std::string &branchName = "", int nBins = 128, Double_t minVal = 0.,
+                                Double_t maxVal = 0.)
+   {
+      auto theBranchName(branchName);
       GetDefaultBranchName(theBranchName, "fill the histogram");
-      auto h = std::make_shared<TH1F>("","",nBins, minVal, maxVal);
-      return CreateAction<T, EActionType::kHisto1D>(theBranchName,h);
+      auto h = std::make_shared<TH1F>("", "", nBins, minVal, maxVal);
+      if (minVal == maxVal) {
+         h->SetCanExtend(TH1::kAllAxes);
+      }
+      return CreateAction<T, Internal::EActionType::kHisto1D>(theBranchName, h);
    }
 
-   template<typename T = double>
-   TActionResultPtr<T> Min(const std::string& branchName = "") {
-      auto theBranchName (branchName);
-      GetDefaultBranchName(theBranchName, "calculate the minimum");
+   template <typename T = Double_t>
+   TActionResultPtr<Double_t> Min(const std::string &branchName = "")
```

Comment:
Wouldn't it be better if `Min`, `Max` and `Mean` returned a `TActionResultPtr` of the same type as the branch instead of always returning `double`? I am not sure how to do it though, we would at least need templated operations.